build/RunTestsOnHelix.sh

```diff
@@ -35,12 +35,4 @@ dotnet nuget remove source dotnet-libraries-transport --configfile $TestExecutio
 dotnet nuget remove source dotnet-tools-transport --configfile $TestExecutionDirectory/NuGet.config
 dotnet nuget remove source dotnet-libraries --configfile $TestExecutionDirectory/NuGet.config
 dotnet nuget remove source dotnet-eng --configfile $TestExecutionDirectory/NuGet.config
-dotnet nuget list source --configfile $TestExecutionDirectory/NuGet.config
-
-cp $HELIX_CORRELATION_PAYLOAD/t/TestExecutionDirectoryFiles/testAsset.props ./
-export TestPackagesRoot=$(pwd)/Assets/TestPackages
-dotnet build ./Assets/TestPackages/Microsoft.NET.TestPackages.csproj /t:Build -p:VersionPropsIsImported=false
-mkdir $TestExecutionDirectory/Testpackages
-cp -v $TestPackagesRoot/TestPackages/* $TestExecutionDirectory/Testpackages/.
-dotnet nuget add source $TestExecutionDirectory/Testpackages --configfile $TestExecutionDirectory/NuGet.config
-
+dotnet nuget list source --configfile $TestExecutionDirectory/NuGet.config
```

Comment:
Why dont we need to add the test execution directory nuget.config anymore?

(dotnet nuget add source $DOTNET_ROOT/.nuget --configfile $TestExecutionDirectory/nuget.config
dotnet nuget list source --configfile)

Reply:
?? It wasn't removed:
![image](https://github.com/dotnet/sdk/assets/17788297/ab5fbf31-93c0-436e-b88c-e7c958442683)

The block of code removed here was building the added .csproj to create these packages. Now, they're just created when Microsoft.NET.TestFramework is built.

Reply:
Interesting, I saw that it was removed in the original PR but wasn't added back in this PR, so that's why I thought it was still removed.